### PR TITLE
Address TODO in code

### LINKED ISSUE #96: Address TODO comments in code
There are a number of TODO comments still to be addressed in the code:

- [x] `scico/blockarray.py:1322`
- [x] `scico/__init__.py:8`
- [x] `scico/numpy/__init__.py:81`
- [x] `scico/loss.py:73`
- [x] `scico/typing.py:29`
- [x] `scico/scipy/special.py:50`

### PR DESCRIPTION
**Pull request by Michael-T-McCann**
closes #96

### CODE REVIEW

File: scico/blockarray.py [hunk 1]
```diff
@@ -289,17 +289,24 @@
 Between BlockArray and :class:`.LinearOperator`
 ***********************************************
 
-.. todo::
-   Improve this
 
 The :class:`.Operator` and :class:`.LinearOperator` classes are designed
-to work on :class:`.BlockArray`. The shapes must conform:
+to work on :class:`.BlockArray`s. For example
```
Comment (bwohlberg) [inline scico/blockarray.py:294]: Perhaps amend slightly to make it clear that they don't *only* work on `BlockArrray`s?

File: scico/blockarray.py [hunk 2]
```diff
@@ -1314,8 +1321,8 @@ def _index_wrapper(self, func_str, values):
         index = self.index
         arr_tuple = self._block_array.split
         if bk_index == Ellipsis:
-            # TODO does this result in multiple copies? One per sub-blockarray,
-            # then one to combine into a nested BA?
+            # This may result in multiple copies: one per sub-blockarray,
```
Comment (bwohlberg) [inline scico/blockarray.py:1324]: This behaviour is confirmed? If not, it would be good to open an issue as a reminder to check.

**Review by bwohlberg — APPROVED** [review]
Approved modulo minor suggestions/questions.

File: scico/blockarray.py (hunk 2 above)
Comment (Michael-T-McCann) [reply scico/blockarray.py:1324]: Issue added #172. I do not fully understand the question nor the code, so it might help if you edited the issue.

Comment (bwohlberg) [reply scico/blockarray.py:1327]: I don't understand the question either. I added a request for comment from @lukepfister to the issue. I think this PR is fine we for now. We can come back to this comment later if the outcome of the discussion in #172 warrants it.

**Pull request merged**